### PR TITLE
Add extensions to pipelined

### LINKED ISSUE #1115: Integrate extensions into pipelined
After #1113 and #1114 are finished, integrate support for extensions into pipelined.

Pipelined should consult the extension registry for handlers and mutators in addition to the internal handlers and mutators that are mapped.

Determine if we need to specify a timeout for RPC so that the pipeline doesn't stall.

Come up with a strategy for how we can communicate failure on RPC errors.

Develop a comprehensive end-to-end test, using the simple handler and mutator extension created in #1114.

### PR DESCRIPTION
**Pull request by echlebek**
## What is this change?

Adds grpc extensions support to pipelined.

## Why is this change necessary?

Closes #1115 

## Were there any complications while making this change?

Filed #1270

### CODE REVIEW

**Comment by echlebek** [comment]
Note that this branch may fail to test cleanly due to #1270

File: backend/pipelined/handle.go [hunk 1]
```diff
@@ -19,6 +19,11 @@ const (
 	DefaultSocketTimeout uint32 = 60
 )
 
+type handlerExtensionUnion struct {
```
Comment (cyphus) [inline backend/pipelined/handle.go]: In the C/C++ universe, the term 'union' means 'two separate types that share the same memory space', I assume it means something different for go? I'm still in learning mode here.

Comment (echlebek) [reply backend/pipelined/handle.go]: Go doesn't have unions, but you can simulate them through anonymous embedding.

**Pull request merged**